### PR TITLE
#3973: Persist scroll position when tab notebooks

### PR DESCRIPTION
**Pull request by rajmusuku**
Fixed the issue of loosing scroll position when we tab away from notebook. Please see below screenshot. I guess we can push it for March release if approved since it will improve user experience.

image

### CODE REVIEW

**Comment by MattIrv** [comment]
A few notes since I've worked on scroll position saving for query results before:
* Tabbing between two of the same type of editor tabs (notebooks) is handled differently than tabbing between two different editor tabs (e.g. notebooks and query editor) so make sure to try out both
* This stuff tends to break frequently especially during VS Code merges. Your code looks cleaner than query editor's but it's worth testing it and thinking about how to test it in a way that is as resilient to VS Code changes as possible

Code looks good to me but I haven't been working on notebooks

**Comment by rajmusuku** [comment]
> A few notes since I've worked on scroll position saving for query results before:
> 
> Tabbing between two of the same type of editor tabs (notebooks) is handled differently than tabbing between two different editor tabs (e.g. notebooks and query editor) so make sure to try out both
> This stuff tends to break frequently especially during VS Code merges. Your code looks cleaner than query editor's but it's worth testing it and thinking about how to test it in a way that is as resilient to VS Code changes as possible
> 
> Code looks good to me but I haven't been working on notebooks

Thanks Matt for your comments. I have tested toggling b/w notebooks and query editors , I don't see any issues. Moreover, since this fix is handled in notebook.component ther may not be any issue with vscode merge as well.

File: src/sql/parts/notebook/notebook.component.ts [hunk 1]
```diff
@@ -157,6 +160,11 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
+	//Saves scrollTop value on scroll change
+	public scrollHandler(event: Event){
+		this._scrollTop = event.srcElement.scrollTop;
```
Comment (chlafreniere) [inline src/sql/parts/notebook/notebook.component.ts]: Does this work when a notebook and a query editor are side by side in split view?

File: src/sql/parts/notebook/notebook.component.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
 <div style="overflow: hidden; width: 100%; height: 100%; display: flex; flex-flow: column">
 	<div #toolbar class="editor-toolbar actionbar-container" style="flex: 0 0 auto; display: flex; flex-flow: row; width: 100%; align-items: center;">
 	</div>
-	<div class="scrollable" style="flex: 1 1 auto; position: relative" (click)="unselectActiveCell()">
+	<div #container class="scrollable" style="flex: 1 1 auto; position: relative" (click)="unselectActiveCell()" (scroll)="scrollHandler($event)">
```
Comment (chlafreniere) [inline src/sql/parts/notebook/notebook.component.html]: We have some (known) scroll issues in notebooks. Does this make those issues worse, better, or is there no change?

Comment (rajmusuku) [reply src/sql/parts/notebook/notebook.component.html]: Yes, it works.

**Pull request merged**